### DISCUSSION TITLE
top level functions for easy jbang one liners

### DISCUSSION THREAD

**Discussion by maxandersen** [jbangdev/jbang#1289] [Ideas]
Been playing with jbang -c lately and seen some constructs that could be easier.

## simple repeat

Having to do foreach is verbose IntStream.range() kinda works:

```
jbang -s faker@jbangdev -c "IntStream.range(1,5)).forEach(i -> println(faker.chuckNorris().fact()))"
```

But maybe we should have `times(int)` method akin groovy `5.times(lambda)`:
```
jbang -s faker@jbangdev -c "times(5).forEach(i -> println(faker.chuckNorris().fact()))"
```

## Argument parsing
Passing arguments to a oneline is tedious:
```
jbang -s faker@jbangdev -c "IntStream.range(1,Integer.valueOf(args[0])).forEach(i -> println(faker.chuckNorris().fact()))"
```
Maybe make a `iargs(pos)`  as shorthand:
```
jbang -s faker@jbangdev -c "IntStream.range(1,iargs(0)).forEach(i -> println(faker.chuckNorris().fact()))"
```

others ?

**Comment by quintesse** [comment]
I find `iargs` to be a bit ... ickly :-)

Perhaps we can do something similar to `Line`? And have a class called `Args` which has the same `s()`, `i()` and `d()` methods and can be accessed using `$args` or something like that.

Another option would be to create toplevel methods `int(String s)` and `int(String s, int defaultValue)` , etc

**Reply by maxandersen** [reply]
Args as Line means you can index them directly...unless you mean `args[0].s()`, `args[0].i()`  that could work but needs to be something else than `args` as that is already the string array.

**Reply by maxandersen** [reply]
something like (as you suggested):

```
$args[0].s()
$args[1].d(23)  //returns 23 if can't be parsed
$args[2].i()
```

**Reply by maxandersen** [reply]
where `$args` is "rich" as opposed to just `args` :)

**Comment by maxandersen** [comment]
Also something to say "give me all lines" as lines() in itself can be too verbose:

```
 http "https://random-word-api.herokuapp.com/word?number=10" | jbang -s jackson.jsh -c "println(json.readTree(lines().collect(Collectors.joining())))"
```
